pkg/alertmanager/firewall.go

```diff
@@ -0,0 +1,67 @@
+package alertmanager
+
+import (
+	"flag"
+
+	"github.com/prometheus/alertmanager/config"
+	commoncfg "github.com/prometheus/common/config"
+
+	"github.com/cortexproject/cortex/pkg/util/flagext"
+	netutil "github.com/cortexproject/cortex/pkg/util/net"
+)
+
+type FirewallConfig struct {
+	BlockCIDRs   flagext.CIDRSliceCSV `yaml:"block_cidrs"`
+	BlockPrivate bool                 `yaml:"block_private_addresses_enabled"`
+}
```

Comment:
I think in the longer term we probably would both need a block and allow approach or even some ordering between the two. I think it might creating a `Block` referencing a separate struct so we can create an Allow in the future easily:

```
type FirewallConfig struct {
  Block FirewallHostsSpec
}
type FirewallHostsSpec struct {
  CIDRS ...
  Private bool
}
```

Reply:
Makes sense. I've done it. What do you think?

Reply:
Yes that looks more reproducible